scrapy/core/downloader/handlers/http11.py

```diff
@@ -121,8 +122,16 @@ def processProxyResponse(self, rcvd_bytes):
         created, notifies the client that we are ready to send requests. If not
         raises a TunnelError.
         """
+        self._connectBuffer += rcvd_bytes
```

Comment:
This could have a different algorithmic complexity in pypy; it seems using bytearray could help. I'm not sure if this is a real problem though.